kubeflow/gcp/setup_backend.sh

```diff
@@ -22,9 +22,18 @@ gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS
 gcloud config list
 
 NODE_PORT=$(kubectl --namespace=${NAMESPACE} get svc ${SERVICE} -o jsonpath='{.spec.ports[0].nodePort}')
+echo "node port is ${NODE_PORT}"
+
+while [[ -z ${BACKENDS} ]]; do
+  BACKENDS=$(kubectl --namespace=${NAMESPACE} get ingress ${SERVICE}-ingress -o jsonpath='{.metadata.annotations.ingress\.kubernetes\.io/backends}')
+  echo "fetching backends info: ${BACKENDS}"
+  sleep 2
+done
+BACKEND_NAME=$(echo $BACKENDS | grep -o "k8s-be-${NODE_PORT}--[0-9a-z]\+")
+echo "backend name is ${BACKEND_NAME}"
 while [[ -z ${BACKEND_ID} ]]; do
-  BACKEND_ID=$(gcloud compute --project=${PROJECT} backend-services list --filter=name~k8s-be-${NODE_PORT}- --format='value(id)')
-  echo "Waiting for backend id PROJECT=${PROJECT} NAMESPACE=${NAMESPACE} SERVICE=${SERVICE} filter=name~k8s-be-${NODE_PORT}- ..."
+  BACKEND_ID=$(gcloud compute --project=${PROJECT} backend-services list --filter=${BACKEND_NAME} --format='value(id)')
```

Comment:
do you need "name~" in filter?

Reply:
I've never checked if not using it - it works with it, though.

Reply:
but you removed the "name~". Do you want to put it back?

Reply:
done.